File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/PreAcademicColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PreAcademicRow } from "./PreAcademicRow";

export interface PreAcademicColumns {
    Id: Column<PreAcademicRow>;
    ExamName: Column<PreAcademicRow>;
    Description: Column<PreAcademicRow>;
    Thumbnail: Column<PreAcademicRow>;
}

export class PreAcademicColumns extends ColumnsBase<PreAcademicRow> {
    static readonly columnsKey = 'Masters.PreAcademic';
    static readonly Fields = fieldsProxy<PreAcademicColumns>();
}

File: GXpert/GXpert.Web/Modules/Masters/PreAcademic/PreAcademicColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.PreAcademic")]
[BasedOnRow(typeof(PreAcademicRow), CheckNames = true)]
public class PreAcademicColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string ExamName { get; set; }
    public string Description { get; set; }
    public string Thumbnail { get; set; }
   
}